index.bs

```diff
@@ -3199,6 +3464,177 @@ The [=remote end steps=] with |command parameters| are:
+
+<div algorithm>
+To <dfn>deserialize an argument value</dfn> given an |serialized argument|:
+
+1. If |serialized argument| is [=RemoteReference=], return [=deserialize a remote reference=] of
+   |serialized argument|.
+
+1. Return [=deserialize a local value=] of |serialized argument|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>deserialize arguments</dfn> given |serialized arguments list|:
+
+1. Let |deserialized arguments list| be an empty list.
+
+1. For each |serialized argument| of |serialized arguments list|:
+
+  1. Let |deserialized argument| be the result of [=try=] to [=deserialize an argument value=] with
+     given |serialized argument|.
+
+  1. If |deserialized argument| is [=error=], return |deserialized argument|.
+
+  1. Append |deserialized argument| to the |deserialized arguments list|.
+
+1. Return [=success=] with data |deserialized arguments list|.
+
+Issue: Clarify behaviour in case of deserialization failed.
+</div>
+
+<div algorithm>
+To <dfn>evaluate function body</dfn> with a given |function declaration|,
+|environment settings|, |base URL|, and |options|:
+
+Note: the |function declaration| is parenthesized and evaluated.
+
+1. Let |function script| be the result of [=create a classic script=] with
+   <code>(</code> + |function declaration| + <code>)</code>, |environment settings|, |base URL|, and |options|.
+
+1. [=Prepare to run script=] with |environment settings|.
+
+1. Let |function body evaluation status| be [=ScriptEvaluation=](|function script|'s record).
+
+1. [=Clean up after running script=] with |environment settings|.
+
+1. Return |function body evaluation status|.
+
+Issue: Define behavior in case of exception during script evaluation.
+</div>
+
+
+The [=remote end steps=] with |command parameters| are:
+
+1. Let |realm| be the result of [=trying=] to [=get a realm from a target=]
+   given the value of the <code>target</code> field of |command parameters|.
+
+1. Let |environment settings| be the [=environment settings object=] whose
+   [=realm execution context=]'s Realm component is |realm|.
+
+1. Let |command arguments| be the value of the <code>arguments</code> field
+   of |command parameters|.
+
+1. Let |deserialized arguments| be the result of [=try=] to [=deserialize arguments=] with given
+   |command arguments|.
+
+1. If |deserialized arguments| is [=error=], return |deserialized arguments|.
+
+1. Let |function declaration| be the value of the
+   <code>functionDeclaration</code> field of |command parameters|.
+
+1. Let |function body evaluation status| be the result of [=evaluate function body=]
+   (|function declaration|, |environment settings|, |base URL|, and |options|).
+
+1. If |function body evaluation status|.\[[Type]] is <code>throw</code>:
+
+  1. Let |exception details| be the result of [=get exception details=] given
+     |function body evaluation status|.
+
+  1. Return a new map matching the <code>ScriptCallFunctionResultException</code>
```

Comment:
I think we should just make passing in something that doesn't successfully evaluate as js a protocol error

Reply:
Return `invalid argument` in case of function evaluation failed.